subsys/dfu/dfu_target/src/dfu_target_modem_delta.c

```diff
@@ -169,83 +115,68 @@ int dfu_target_modem_delta_init(size_t file_size, dfu_target_callback_t cb)
 		return -EFBIG;
 	}
 
-	/* Check offset, store to local variable */
-	err = getsockopt(fd, SOL_DFU, SO_DFU_OFFSET, &offset, &len);
-	if (err < 0) {
-		if (errno == ENOEXEC) {
-			LOG_ERR("Modem error: %d", get_modem_error());
-		} else {
-			LOG_ERR("getsockopt(OFFSET) errno: %d", errno);
-		}
+	/* Check offset and erase firmware if necessary */
+	err = nrf_modem_delta_dfu_offset(&offset);
+	if (err != 0) {
+		LOG_ERR("Failed to retrieve offset in scratch area, error %d", err);
+		return -EFAULT;
 	}
 
 	if (offset == DIRTY_IMAGE) {
```

Comment:
You could use the define that comes with the API now: `NRF_MODEM_DELTA_DFU_OFFSET_DIRTY`